### ISSUE TITLE
Command to Run TPT and C-TPT

### ISSUE THREAD

**Issue by ashshaksharifdeen** [Bala93/CLIPCalib#1]
Hi,

I would like to explore your methods and reproduce the results. However, I couldn't find the command to reproduce the results for TPT and C-TPT. Is there a specific command or initialization I need to use?

**Comment by Bala93** [comment]
https://github.com/Bala93/CLIPCalib/tree/test-time-prompt Please check this branch

**Comment by ashshaksharifdeen** [comment]
Yes, I have checked it with the following two commands for the SaLs method on the DTD dataset:

bash scripts/adapt.sh 0 imagenet SGD_lr1e-1_B256_ep300 1 TR none RN50
bash scripts/eval.sh 0 imagenetv2 SGD_lr1e-1_B256_ep300 1 TR none RN50

I am getting:

Accuracy on test: 42.79
Calibration on test: ECE, CECE: 4.05, 1.08
I am just skeptical about whether I am doing it correctly or not because the results do not match the results mentioned in the paper for C-TPT method.

**Comment by Bala93** [comment]
Those commands are for checking the adapters based methods. You can refer to TPT repository.

**Comment by ashshaksharifdeen** [comment]
Ok. Thanks For your Clarification

**Comment by ashshaksharifdeen** [comment]
Hi,

In the TPT-main folder, within files like ctpt_classification_constrained.py and other main method files, you imported ECELoss and ClasswiseECELoss from utils.metrics. However, there is no metrics file under the utils folder to compute ECE and CECE, which caused a "no module" error. Therefore, I used the code from the utils files in the main branch to compute ECE and CECE, but I got mismatched results. Do you have a specific metrics file under the utils folder of TPT-main?

**Comment by Bala93** [comment]
In all the three methods I saved the logits and obtained the calibration metrics separately.

**Comment by ashshaksharifdeen** [comment]
How did you do that? Do you have specific code for it in your repository, or how did you accomplish it?

**Comment by Bala93** [comment]
You can find calib summary code in repo. For saving logits, it should be already there in the training code.

**Comment by ashshaksharifdeen** [comment]
But that calib_summary.py file is written to check adapter methods, right? However, you also used metrics from it, but I couldn't find the metric file. Specifically, the line from metrics import expected_calibration_error, maximum_calibration_error, AdaptiveECELoss.

**Comment by Bala93** [comment]
You can also use that code, by giving the path of logit outputs from tpt

**Comment by Bala93** [comment]
Thanks for pointing out the missing metric file, will add it in some time.

**Comment by Bala93** [comment]
Added the file, let me know if it works now.

**Comment by ashshaksharifdeen** [comment]
Thanks for updating it. Yes it works
